src/TemplateStudio/Utils.js

```diff
@@ -0,0 +1,198 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+/* Libraries */
+
+import moment from 'moment'; // For DateTime support during contract execution
+
+/* Ergo */
+
+import Ergo from '@accordproject/ergo-compiler/lib/ergo.js';
+
+function getUrlVars() {
+    let vars = {};
+    window.location.href.replace(/[?&]+([^=&]+)=([^&]*)/gi, function(m,key,value) {
+        vars[key] = decodeURIComponent(value);
+    });
+    return vars;
+}
+function getUrlParam(parameter, defaultvalue){
+    let urlparameter = defaultvalue;
+    try {
+        if(window.location.href.indexOf(parameter) > -1){
+            if (getUrlVars()[parameter]) {
+                urlparameter = getUrlVars()[parameter];
+            } else {
+                throw new Error('Cannot parse template URL parameter');
+            }
+        } else {
+            console.log('Did not find template URL parameter when loading studio, using default: ' + urlparameter);
+        }
+    } catch (error) {
+        console.log('Did not find template URL parameter when loading studio, using default: ' + urlparameter);
+    }
+    return urlparameter;
+}
+function initUrl(defaultUrl) {
+    return getUrlParam('template', defaultUrl);
+}
+function parseSample(input_state,text) {
+    const state = input_state;
+    const clause = input_state.clause;
+    try {
+        clause.parse(text);
+        state.data = JSON.stringify(clause.getData(),null,2);
+        state.log.text = 'Parse successful!',
+        state.text = text;
+    } catch (error){
+        state.data = 'null';
+        state.log.text = '[Parse Contract] ' + error.message;
+        state.text = text;
+    }
+    return state;
+}
+
+function generateText(input_state,data) {
+    const state = input_state;
+    try {
+        const dataContent = JSON.parse(data);
+        const clause = input_state.clause;
+        clause.setData(dataContent);
+        const text = clause.generateText();
+        state.text = text;
+        state.data = data;
+        if (updateSample(clause,text)) {
+            state.status = 'changed';
+        }
+        state.log.text = 'GenerateText successful!';
+    } catch (error){
+        state.data = data;
+        state.log.text = '[Instantiate Contract] ' + error.message;
+    }
+    return state;
+}
+
+function compileLogic(editor,logic,state) {
```

Comment:
the functions that are taking the state object as a parameter and then mutating it should still be methods on the component itself, since we are eventually going to want to fix the places where we directly mutate the state to use `this.setState` instead.